src/cascadia/TerminalCore/ColorFix.cpp

```diff
@@ -0,0 +1,280 @@
+    double var_X = A / 500. + var_Y;
+    double var_Z = var_Y - B / 200.;
+
+    if (pow(var_Y, 3) > 0.008856)
+        var_Y = pow(var_Y, 3);
+    else
+        var_Y = (var_Y - 16. / 116.) / 7.787;
+    if (pow(var_X, 3) > 0.008856)
+        var_X = pow(var_X, 3);
+    else
+        var_X = (var_X - 16. / 116.) / 7.787;
+    if (pow(var_Z, 3) > 0.008856)
+        var_Z = pow(var_Z, 3);
+    else
+        var_Z = (var_Z - 16. / 116.) / 7.787;
+
+    double X = 95.047 * var_X; //ref_X =  95.047     (Observer= 2 degrees, Illuminant= D65)
+    double Y = 100.000 * var_Y; //ref_Y = 100.000
+    double Z = 108.883 * var_Z; //ref_Z = 108.883
+
+    var_X = X / 100.; //X from 0 to  95.047      (Observer = 2 degrees, Illuminant = D65)
+    var_Y = Y / 100.; //Y from 0 to 100.000
+    var_Z = Z / 100.; //Z from 0 to 108.883
+
+    double var_R = var_X * 3.2406 + var_Y * -1.5372 + var_Z * -0.4986;
+    double var_G = var_X * -0.9689 + var_Y * 1.8758 + var_Z * 0.0415;
+    double var_B = var_X * 0.0557 + var_Y * -0.2040 + var_Z * 1.0570;
+
+    if (var_R > 0.0031308)
+        var_R = 1.055 * pow(var_R, (1 / 2.4)) - 0.055;
+    else
+        var_R = 12.92 * var_R;
+    if (var_G > 0.0031308)
+        var_G = 1.055 * pow(var_G, (1 / 2.4)) - 0.055;
+    else
+        var_G = 12.92 * var_G;
+    if (var_B > 0.0031308)
+        var_B = 1.055 * pow(var_B, (1 / 2.4)) - 0.055;
+    else
+        var_B = 12.92 * var_B;
+
+    r = _Clamp(var_R * 255.);
+    g = _Clamp(var_G * 255.);
+    b = _Clamp(var_B * 255.);
+}
+
+// Method Description:
+// - Given foreground and background colors, change the foreground color to
+//   make it more perceivable if necessary
+// - Arguments:
+// - fg: the foreground color
+// - bg: the background color
+// - Return Value:
+// - The foreground color after performing any necessary changes to make it more perceivable
+COLORREF ColorFix::GetPerceivableColor(COLORREF fg, COLORREF bg)
+{
+    // If the colors are the same, don't do any adjusting
+    if (fg == bg)
+    {
+        return fg;
+    }
+    ColorFix backLab(bg);
+    ColorFix frontLab(fg);
+    double de1 = GetDeltaE(frontLab, backLab);
+    if (de1 < gMinThreshold)
+    {
+        for (int i = 0; i <= 1; i++)
+        {
+            double step = (i == 0) ? gLStep : -gLStep;
+            frontLab.L += step;
+
+            while (((i == 0) && (frontLab.L <= 100)) || ((i == 1) && (frontLab.L >= 0)))
+            {
+                double de2 = GetDeltaE(frontLab, backLab);
+                if (de2 >= gExpThreshold)
+                {
+                    frontLab._ToRGB();
+                    return frontLab.rgb;
+                }
+                frontLab.L += step;
+            }
+        }
+    }
+    return frontLab.rgb;
+}
+
+// Method Description:
+// - Clamps the given value to be between 0-255 inclusive
+// - Converts the result to BYTE
+// Arguments:
+// - v: the value to clamp
+// Return Value:
+// - The clamped value
+BYTE ColorFix::_Clamp(double v)

```

Comment:
can we use `std::clamp` here? Or even remove the function entirely and use `std::clamp`?